tests/run.sh

```diff
@@ -18,7 +18,7 @@ exec &> >(tee -a "$CLUSTER_NAME.log")
 
 function destroy() {
   echo -e "\\e[34m Exiting... Destroying Tectonic...\\e[0m"
-  tectonic destroy --dir="${CLUSTER_NAME}"
+  "${TECTONIC}" destroy --dir="${CLUSTER_NAME}"
```

Comment:
This doesn't seem to work either:

```console
$ git describe --always --dirty
2d3306a-dirty
$ git diff
diff --git a/tests/run.sh b/tests/run.sh
index 407e594..22373c0 100755
--- a/tests/run.sh
+++ b/tests/run.sh
@@ -26,6 +26,7 @@ function destroy() {
 trap destroy EXIT
 
 echo -e "\\e[36m Starting build process...\\e[0m"
+exit  # FIXME: just for testing
 bazel build tarball smoke_tests
 # In future bazel build could be extracted to another job which could be running in docker container like this:
 # docker run --rm -v $PWD:$PWD:Z -w $PWD quay.io/coreos/tectonic-builder:bazel-v0.3 bazel build tarball smoke_tests
$ PULL_SECRET_PATH=~/src/openshift/installer/pull-secret.json tests/run.sh 
 Starting build process...
 Exiting... Destroying Tectonic...
tests/run.sh: line 21: : command not found
```

Things that do work:

```diff
diff --git a/tests/run.sh b/tests/run.sh
index 407e594..f74d94f 100755
--- a/tests/run.sh
+++ b/tests/run.sh
@@ -16,6 +16,8 @@ SMOKE_TEST_OUTPUT="Never executed. Problem with one of previous stages"
 CLUSTER_NAME=$(echo "${PREFIX}-$(uuidgen -r | cut -c1-5)" | tr '[:upper:]' '[:lower:]')
 exec &> >(tee -a "$CLUSTER_NAME.log")
 
+TECTONIC="${PWD}/tectonic-dev/installer/tectonic"
+
 function destroy() {
   echo -e "\\e[34m Exiting... Destroying Tectonic...\\e[0m"
   "${TECTONIC}" destroy --dir="${CLUSTER_NAME}"
@@ -26,6 +28,7 @@ function destroy() {
 trap destroy EXIT
 
 echo -e "\\e[36m Starting build process...\\e[0m"
+exit  # FIXME: just for testing
 bazel build tarball smoke_tests
 # In future bazel build could be extracted to another job which could be running in docker container like this:
 # docker run --rm -v $PWD:$PWD:Z -w $PWD quay.io/coreos/tectonic-builder:bazel-v0.3 bazel build tarball smoke_tests
@@ -33,7 +36,6 @@ bazel build tarball smoke_tests
 echo -e "\\e[36m Unpacking artifacts...\\e[0m"
 tar -zxf bazel-bin/tectonic-dev.tar.gz
 cp bazel-bin/tests/smoke/linux_amd64_stripped/go_default_test tectonic-dev/smoke
-export TECTONIC="${PWD}/tectonic-dev/installer/tectonic"
 cd tectonic-dev
 
 ### HANDLE SSH KEY ###
```

or

```diff
diff --git a/tests/run.sh b/tests/run.sh
index 407e594..236de0e 100755
--- a/tests/run.sh
+++ b/tests/run.sh
@@ -23,8 +23,6 @@ function destroy() {
   echo -e "\\e[34m So Long, and Thanks for All the Fish\\e[0m"
 }
 
-trap destroy EXIT
-
 echo -e "\\e[36m Starting build process...\\e[0m"
 bazel build tarball smoke_tests
 # In future bazel build could be extracted to another job which could be running in docker container like this:
@@ -33,9 +31,12 @@ bazel build tarball smoke_tests
 echo -e "\\e[36m Unpacking artifacts...\\e[0m"
 tar -zxf bazel-bin/tectonic-dev.tar.gz
 cp bazel-bin/tests/smoke/linux_amd64_stripped/go_default_test tectonic-dev/smoke
-export TECTONIC="${PWD}/tectonic-dev/installer/tectonic"
 cd tectonic-dev
 
+export TECTONIC="${PWD}/installer/tectonic"
+trap destroy EXIT
+exit  # FIXME: just for testing
+
 ### HANDLE SSH KEY ###
 if [ ! -f ~/.ssh/id_rsa.pub ]; then
   echo -e "\\e[36m Generating SSH key-pair...\\e[0m"
```

or

```diff
diff --git a/tests/run.sh b/tests/run.sh
index 407e594..487b76f 100755
--- a/tests/run.sh
+++ b/tests/run.sh
@@ -18,13 +18,11 @@ exec &> >(tee -a "$CLUSTER_NAME.log")
 
 function destroy() {
   echo -e "\\e[34m Exiting... Destroying Tectonic...\\e[0m"
-  "${TECTONIC}" destroy --dir="${CLUSTER_NAME}"
+  installer/tectonic destroy --dir="${CLUSTER_NAME}"
   echo -e "\\e[36m Finished! Smoke test output:\\e[0m ${SMOKE_TEST_OUTPUT}"
   echo -e "\\e[34m So Long, and Thanks for All the Fish\\e[0m"
 }
 
-trap destroy EXIT
-
 echo -e "\\e[36m Starting build process...\\e[0m"
 bazel build tarball smoke_tests
 # In future bazel build could be extracted to another job which could be running in docker container like this:
@@ -33,9 +31,11 @@ bazel build tarball smoke_tests
 echo -e "\\e[36m Unpacking artifacts...\\e[0m"
 tar -zxf bazel-bin/tectonic-dev.tar.gz
 cp bazel-bin/tests/smoke/linux_amd64_stripped/go_default_test tectonic-dev/smoke
-export TECTONIC="${PWD}/tectonic-dev/installer/tectonic"
 cd tectonic-dev
 
+trap destroy EXIT
+exit  # FIXME
+
 ### HANDLE SSH KEY ###
 if [ ! -f ~/.ssh/id_rsa.pub ]; then
   echo -e "\\e[36m Generating SSH key-pair...\\e[0m"
@@ -91,10 +91,10 @@ python <<-EOF >"${CLUSTER_NAME}.yaml"
 	EOF
 
 echo -e "\\e[36m Initializing Tectonic...\\e[0m"
-"${TECTONIC}" init --config="${CLUSTER_NAME}".yaml
+installer/tectonic init --config="${CLUSTER_NAME}".yaml
 
 echo -e "\\e[36m Deploying Tectonic...\\e[0m"
-"${TECTONIC}" install --dir="${CLUSTER_NAME}"
+installer/tectonic install --dir="${CLUSTER_NAME}"
 echo -e "\\e[36m Running smoke test...\\e[0m"
 export SMOKE_KUBECONFIG="${PWD}/${CLUSTER_NAME}/generated/auth/kubeconfig"
 export SMOKE_NODE_COUNT="5"  # Sum of all nodes (master + worker)
```

Reply:
I moved the variable to the others in the front